### PR TITLE
Manage JS libraries using npm and webpack.

### PR DESCRIPTION
**Pull request by awong-dev**
Use npm's package.json to manage all the vendor javascript library versioning. Then use webpack to bundle them for our site's usage.

This will allow us to clean up our asset/js directory so that it only contains code we write. It'll also give us version and dependency management on the vendor libraries.

During the audit process, it turns out only a few of the checked in Javascript libraries were being used. Those have been either ported, or in the case of classie.js, replaced with an HTML5 polyfill.

In package.json, the devDependencies are reserved for things that build the libraries and the dependencies section is used to contain the libraries themselves. This break up is arbitrary because webpack always needs to install both buckets, but it makes it a little easier to read.  The one exception to this rule is modernizr because modernizer is, technically, not a standalone library but rather a custom feature test builder. Thus, both modernizr and modernizr-loader are in devDependencies.

Webpack was chosen over bower because the npm ecosystem with webpack seems to have overtaken bower in size and developer mindshare.

This obsoletes #692.

### CODE REVIEW

**Comment by awong-dev** [comment]
@shanear @emilyville @dannychapman

I could use a review here.

File: _includes/header.html [hunk 1]
```diff
@@ -51,24 +51,6 @@
 {% endif %}
 
 
-<!-- Header Javascript and CSS for animation-->
-<script src="/assets/js/vendor/jquery-1.11.0.min.js"></script>
-<script src="/assets/js/components.js"></script>
-<script src="/assets/js/vendor/modernizr.custom.js"></script>
-<script src="/assets/js/vendor/wow.min.js"></script>
```
Comment (shanear) [inline _includes/header.html]: Should these files also be removed if we're not using vendored versions of them anymore

**Comment by shanear** [comment]
Should `rake serve` now run `webpack --watch` as well, to make sure js changes are auto-built?

**Comment by awong-dev** [comment]
It should, but I didn't know how rake task management worked so left that out for now.

File: _includes/header.html (hunk 1 above)
Comment (awong-dev) [inline _includes/header.html]: yes, but I'd like to do in another PR after we've verified webpack works.

**Comment by awong-dev** [comment]
@shanear can I get an LGTM for the webpack bits?

As for whether or not it fits correctly into the vets.gov architecture, I think I'll take responsibility for evaluating that one.

**Comment by shanear** [comment]
LGTM

**Comment by awong-dev** [comment]
per notification on #vetsdotgov, I'm going to merge this now. If anyone objects, we can still revert pretty easily.
